### PR TITLE
Add `getLayers`, `getWmsLayer`, `getWmtsLayer` and `createWmtsStore`

### PR DESCRIPTION
**Pull request by JakobMiksch**
Adds additional functions

### CODE REVIEW

File: src/datastore.js [hunk 1]
```diff
@@ -378,6 +378,41 @@ export default class DatastoreClient {
     }
   }
 
+  /**
+   * Creates a WMTS based data store.
+   *
+   * @param {String} workspace The WS to create the data store in
+   * @param {String} dataStore The data store name
+   * @param {String} wmsCapabilitiesUrl Base WMTS capabilities URL
```
Comment (chrismayer) [inline src/datastore.js:386]:
```suggestion
   * @param {String} wmtsCapabilitiesUrl Base WMTS capabilities URL
```

File: src/layer.js [hunk 2]
```diff
@@ -133,6 +133,103 @@ export default class LayerClient {
...
+  /**
+   * Returns information about a cascaded WMS layer.
+   *
+   * @param {String} workspace The workspace
+   * @param {String} datastore The datastore
+   * @param {String} layerName The WMS layer name
+   *
+   * @throws Error if request fails
+   *
+   * @returns {Object} An object with layer information or undefined if it cannot be found
```
Comment (chrismayer) [inline src/layer.js:171]: Shouldn't it be a HTTP 404 in case it cannot be found?

File: src/layer.js [hunk 3]
```diff
@@ -133,6 +133,103 @@ export default class LayerClient {
...
+  /**
+   * Returns information about a cascaded WMTS layer.
+   *
+   * @param {String} workspace The workspace
+   * @param {String} datastore The datastore
+   * @param {String} layerName The WMTS layer name
+   *
+   * @throws Error if request fails
+   *
+   * @returns {Object} An object with layer information or undefined if it cannot be found
```
Comment (chrismayer) [inline src/layer.js:207]: see above...

File: test/test.js [hunk 4]
```diff
@@ -194,6 +194,15 @@ describe('Datastore', () => {
       wmsUrl);
   });
 
+  it('can create a WMTS Store', async () => {
+    // TODO: make sure the WMS actually exists
```
Comment (chrismayer) [inline test/test.js:198]: Copy/paste? can this be removed?

File: test/test.js [hunk 5]
```diff
@@ -383,6 +393,26 @@ describe('Layer', () => {
...
   });
 
+  it('can get layers by workspace', async () => {
+    const result = await grc.layers.getLayers(workSpace);
+    expect(result.layers.layer.length).to.equal(3);
+  });
+
+  it('works with non-existing workspaces', async () => {
+    const result = await grc.layers.getLayers('fantasy-workspace');
+    expect(result.layers).to.equal('');
```
Comment (chrismayer) [inline test/test.js:403]: Is this correct by having an empty string (`''`) as expectation value?

File: demo/index.js [hunk 6]
```diff
@@ -226,6 +226,21 @@ async function main () {
     //   prettyJson(await grc.layers.getCoverage(ws, 'testCovStore', 'testCoverage'))
     // );
 
+    // console.log(
+    //   'Get Layers',
+    //   prettyJson(await grc.layers.getLayers('asdf'))
```
Comment (chrismayer) [inline demo/index.js:231]: Better anything meaningful from above instead of `'asdf'`?

File: demo/index.js [hunk 7]
```diff
@@ -226,6 +226,21 @@ async function main () {
...
+    // );
+
+    // console.log(
+    //   'Get WMS Layer',
+    //   prettyJson(await grc.layers.getWmsLayer('asdf', 'meg', 'ne_worldX' ))
```
Comment (chrismayer) [inline demo/index.js:236]: Better anything meaningful from above instead of `'asdf'`?

**Review by chrismayer — CHANGES_REQUESTED** [review]
Looks good in general, thanks for this @JakobMiksch. Some minor things, which should be clarified before merging.

File: test/test.js (hunk 5 above)
Comment (JakobMiksch) [reply test/test.js:403]: true. fixed in 30c30d4

File: demo/index.js (hunk 7 above)
Comment (JakobMiksch) [reply demo/index.js]: true. Fixed in 30c30d4

File: test/test.js (hunk 4 above)
Comment (JakobMiksch) [reply test/test.js:198]: This note shall be a reminder, that the provided GeoServer might be down and therefore the tests fail. But we can also convert this to an issue.

File: src/layer.js (hunk 2 above)
Comment (JakobMiksch) [reply src/layer.js:171]: Returning undefined is done intentionally if someone wants to check if a layer a exists or not. The same behavior is done in `getFeatureType`, `getCoverage` and `getWmsLayer`. The function only throws an error if GeoServer cannot be reached at all. 
Is this okay for you?

File: src/layer.js (hunk 3 above)
Comment (JakobMiksch) [reply src/layer.js:207]: see response above ...

**Comment by JakobMiksch** [comment]
@chrismayer thanks for your review. I addressed your remarks and also commented on questions.

File: src/layer.js (hunk 2 above)
Comment (chrismayer) [reply src/layer.js:171]: OK for consistency

File: test/test.js (hunk 4 above)
Comment (JakobMiksch) [reply test/test.js]: true, I fixed it with e232d88

File: src/layer.js (hunk 3 above)
Comment (JakobMiksch) [reply src/layer.js:207]: closing, as reponse above is closed as well https://github.com/meggsimum/geoserver-node-client/pull/91/files#r922013769

**Review by chrismayer — APPROVED** [review]
OK looks good to me now. Feel free to merge @JakobMiksch . Thanks again.

**Pull request merged**